nautobot/docs/plugins/development.md

```diff
@@ -609,6 +610,89 @@ secrets_providers = [ConstantValueSecretsProvider]
+* Creation of a filter on a field that does not currently have filtering support
+* Convenience methods for highly nested fields
+
+There are several conditions that must be met in order to extend a filter:
+
+* The original FilterSet must follow the pattern: `f"{model.__name__}FilterSet"` e.g. `TenantFilterSet`
+* The `PluginFilterExtension.filterset_fields` attribute must be a valid dict, with each key being the filter name (which must start with the plugin's `name` + `_`, e.g. `"example_plugin_description"`, not merely `"description"`) and each value being a valid [django-filter](https://django-filter.readthedocs.io/en/main/) filter
+
+Nautobot will dynamically generate the additional relevant lookup expressions of a plugin's defined custom FilterSet field, so no need to additionally register `example_plugin_description__ic`, etc.
+
+Similar to FilterSet fields, Nautobot provides a default filter form for each model, however that does not cover every possible use case. To list a few examples of why one may want to extend a filter form:
+
+* The base filter form does not include a custom filter defined by the plugin as described above
+* The base filter form does not provide a specific lookup expression to a filterable field, such as allowing regex on name
+
+There are several conditions that must be met in order to extend a filter:
+
+* The original FilterForm must follow the pattern: `f"{model.__name__}FilterForm"`, e.g. `TenantFilterForm`
+* The `filterform_fields` attribute must be a valid dictionary of Django form fields
+
+!!! note
+    A plugin is not required to define both `filterset_fields` and `filterform_fields`.
+
+Example:
+```python
+# filter_extensions.py
+from django import forms
+
+from nautobot.extras.plugins import PluginFilterExtension
+from nautobot.utilities.filters import MultiValueCharFilter
+
+
+def suffix_search(queryset, name, value):
+    return queryset.filter(description=f"{value[0]}.nautobot.com")
+
+
+class TenantFilterExtension(PluginFilterExtension):
+    model = "tenancy.tenant"
+
+    filterset_fields = {
+        "example_plugin_description": MultiValueCharFilter(field_name="description", label="Description"),
+        "example_plugin_sdescrip": MultiValueCharFilter(
+            field_name="description", label="Description", method=suffix_search
+        ),
+        "example_plugin_dtype": MultiValueCharFilter(
+            field_name="sites__devices__device_type__slug", label="Device Type"
+        ),
+    }
+
+    filterform_fields = {
+        "example_plugin_description": forms.CharField(required=False, label="Description"),
+        "example_plugin_dtype": forms.CharField(required=False, label="Device Type"),
+        "slug__ic": forms.CharField(required=False, label="Slug Contains"),
+        "example_plugin_sdescrip": forms.CharField(required=False, label="Suffix Description"),
+    }
+
+
+filter_extensions = [TenantFilterExtension]
+```
```

Comment:
I don't like seeing this code duplicated in the documentation. If we decide to update it, now it has to be maintained in two places. It would be ideal if we could include raw source instead since this is all built from file paths? The `example_plugin` is found outside of the `nautobot` source tree so it might not be feasible today, but I would like this to be considered for the future.

Reply:
Left as is until a pattern is defined.